articles/most-useful-node.js-packages/index.md

```diff
@@ -0,0 +1,515 @@
+changes to your code or method of development. It is a replacement
+wrapper for the node. Nodemon is simple to use, you just need to replace
+the word node with `nodemon` on the command-line when executing your
+`script`. Ie
+`nodemon ./server.js localhost 300`
+
+**Example of other monitoring Frameworks include**
+
+-   [`PM2`](https://www.npmjs.com/package/pm2) - Node.js production
+    process manager that allows you to keep applications alive forever
+    and reload without downtime to facilitate common system admin task.
+-   [`Trace`](https://github.com/RisingStack/trace-nodejs) - Designed for
+    microservices visualized stack trace platform.
+-   [`Forever`](https://www.npmjs.com/package/forever) - CLI tool for
+    ensuring that a given Node.js script runs continuously ie forever.
+
+### 12. Restify
+
+It is a Node.js web service framework, which is optimized for building
+semantic RESTful web services (API) that are ready for production use at
+scale.
+
+**Why you should use Restify**
+
+-   **Production-ready:** Optimizes for introspection and performance.
+-   **Debuggable**: Trace problems back to the origin. It is built from
+    a post-mortem debugging perspective.
+-   **Semantically correct:** It has references littered all-over GitHub
+    and the codebase.
+
+**Examples of simple Restify practice**
+
+#### i. SERVER
+```js
+var restify = require('restify');
+const server = restify.createServer({
+    name: 'myapp',
+    version: '1.0.0'
+});
+server.use(restify.plugins.acceptParser(server.acceptable));
+server.use(restify.plugins.queryParser());
+server.use(restify.plugins.bodyParser());
+server.get('/echo/:name', function (req, res, next) {
+    res.send(req.params);
+    return next();
+});
+server.listen(8080, function () {
+    console.log('%s listening at %s', server.name, server.url);
+});
+```
+#### ii. CLIENT
+```js
+var assert = require('assert');
+var clients = require('restify-clients');
+var client = clients.createJsonClient({
+    url: 'http://localhost:8080',
+    version: '~1.0'
+});
+client.get('/echo/mark', function (err, req, res, obj) {
+    assert.ifError(err);
+    console.log('Server returned: %j', obj);
+});
+```
+#### iii. ROUTING
+```js
+function send(req, res, next) {
+    res.send('hello ' + req.params.name);
+    return next();
+}
+server.post('/hello', function create(req, res, next) {
+    res.send(201, Math.random().toString(36).substr(3, 8));
+    return next();
+});
+server.put('/hello', send);
+server.get('/hello/:name', send);
+server.head('/hello/:name', send);
+server.del('hello/:name', function rm(req, res, next) {
+    res.send(204);
+    return next();
+});
+```
+**Other Node.js API frameworks include:**
+
+-   [`Actionhero`](http://www.actionherojs.com/) - API framework for both
+    TCP sockets, web socket, and HTTP client that create reusable and
+    scalable APIs.
+-   [`FeathersJS`](http://feathersjs.com/) - A framework REST APIS and
```

Comment:
And the same with this one. You need to check all links go to the https version if there is one.